src/test/java/seedu/address/model/student/StudentTest.java

```diff
@@ -92,4 +101,64 @@ public void toStringMethod() {
                 + ALICE.getAssignmentList() + ", remark=" + ALICE.getRemark() + "}";
         assertEquals(expected, ALICE.toString());
     }
+
+    @Test
+    public void hasAssignment() {
+        Student studentWithAssignment = new StudentBuilder(ALICE)
+                .withAssignments(List.of(MATH_HOMEWORK))
+                .buildWithAssignment();
+
+        System.out.println(studentWithAssignment);
+        // same assignment -> returns true
+        assertTrue(studentWithAssignment.hasAssignment(MATH_HOMEWORK));
+
+        // different assignment -> returns false
+        assertFalse(studentWithAssignment.hasAssignment(PHYSICS_HOMEWORK));
+    }
+
+    @Test
+    public void addAssignment() {
+        Student student = new StudentBuilder(ALICE).build();
+
+        // add assignment -> assignment should be in assignment list
+        Student studentWithAssignment = student.addAssignment(MATH_HOMEWORK);
+        assertTrue(studentWithAssignment.hasAssignment(MATH_HOMEWORK));
+        // original student remains unchanged
+        assertFalse(student.hasAssignment(MATH_HOMEWORK));
+    }
+
+    @Test
+    public void deleteAssignment_validIndex() {
+        List<Assignment> assignments = new ArrayList<Assignment>(Arrays.asList(MATH_HOMEWORK, PHYSICS_HOMEWORK));
+        Student student = new StudentBuilder(ALICE).withAssignments(assignments).buildWithAssignment();
+
+        // delete first assignment
+        student.deleteAssignment(1);
+
+        // assignment list should contain only the second assignment
+        assertEquals(1, student.getAssignmentList().size());
+        assertEquals(PHYSICS_HOMEWORK, student.getAssignmentList().get(0));
+    }
+
+    @Test
+    public void deleteAssignment_invalidIndex_throwsAssignmentIndexOutOfRangeException() {
+        Student student = new StudentBuilder(ALICE).withAssignments(
+                new ArrayList<Assignment>(Arrays.asList(MATH_HOMEWORK, PHYSICS_HOMEWORK))).buildWithAssignment();
+
+        // invalid index -> throws AssignmentIndexOutOfRangeException
+        assertThrows(AssignmentIndexOutOfRangeException.class, () -> student.deleteAssignment(3));
+        assertThrows(AssignmentIndexOutOfRangeException.class, () -> student.deleteAssignment(0));
```

Comment:
I wonder if it will be better to make all of these integer values constants in CommandTestUtil.

Reply:
I'll put them as constants but just in this class, since they are specific to this particular test case